servicetalk-transport-api/src/main/java/io/servicetalk/transport/api/ConnectExecutionStrategy.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright © 2018 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.transport.api;
+
+/**
+ * An execution strategy for creating or accepting connections.
+ */
+public interface ConnectExecutionStrategy extends ExecutionStrategy {
+
+    /**
+     * Returns {@code true} if the instance has offloading for any operation.
+     *
+     * @return {@code true} if the instance has offloading for any operation.
+     */
+    @Override
+    default boolean hasOffloads() {
+        return isConnectOffloaded();
+    }
+
+    /**
+     * Returns true if connection creation or accept requires offloading.
+     *
+     * @return true if connection creation or accept requires offloading
+     */
+    boolean isConnectOffloaded();
```

Comment:
Did you consider promoting this flag to `ExecutionStrategy`. Understand that `HttpExecutionStrategy` will also inherit it and it will have no influence on HTTP filters. However, it might be useful for users to specify it on the client/server builder to explicitly tell if they need offloading for connect or not.

Reply:
I have a lot of concerns about flags being present in situations where they aren't used and similarly about the implications of adding flags in the future that existing usages don't contemplate. It is not always clear that a new flag with appropriate default behavior can be cleanly added. It seems that in a lot of cases the default for the new flag, and the absence of providing an appropriate response for the flag by existing code could result in changes in behaviour (resulting from the application of the default) just by introducing a new flag.

I have tried to keep the flags very specific to the context in which they are used so that new flags can be safely added and implementors don't have to worry about affecting behaviour for features they are not using.